design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Visual Query Builder
+description: 
+published: true
+date: 2022-03-08 18:02:53
+tags: 
+editor: markdown
+dateCreated: 2022-03-08 18:02:58
+---
+
+## Context
+
+This document specifies the design for a visual query builder named `Data Explorer` within the Mathesar application. Through `Data Explorer`, users will be able to analyze data across one or multiple tables of a schema and save the resulting tables as views.
+
+Additionally, users will be able to open existing views in `Data Explorer`, as long as its query commands and functions are supported.
+
+## Scenarios
+
+## Starting Points
+
+`Data Explorer` is meant to be a part of multiple flows like the creation of new views, the inspection of existing ones, and as a way to explore data for goals beyond view creation. For this reason, different starting points are considered from which `Data Explorer` might be accessed.
+
+### User opens the Data Explorer to create a new view
+
+#### From Scratch (Top Navigation)
+
+- In the `Schema Browser` view, a user clicks on the `Data Explorer` navigation item located in the top navigation bar.
+- `Data Explorer` opens, and the layout changes. The schema explorer sidebar is hidden, as well as any open table.
+
+#### From a Table (Sidebar Navigation or Open Table)
```

Comment:
@ghislaineguerin I think this scenario is out of scope for this issue, there are many details to consider here. I made a separate issue for this, see: https://github.com/centerofci/mathesar/issues/782

Secondly, we need to make the distinction between queries and views more apparent. Anything opened in the data explorer is still at the query stage, it has not yet been saved as a view. If we have a button called "Create View", then the user will expect us to have created a view. If we just want to open the table in the data explorer, then we should name the button differently. But we can handle all of that in https://github.com/centerofci/mathesar/issues/782, it seems better to remove this scenario and keep this spec focused on the query builder interface rather than how to get to it. Let me know if you disagree.